### ISSUE TITLE
Try canceling TS Server geterr requests when document changes

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/vscode#49251]
From https://github.com/Microsoft/TypeScript/issues/19458#issuecomment-385810670

TS would like us to try cancelling the TS Server geterr requests if the document changes before a response is returned

**Comment by mjbvz** [comment]
See https://github.com/Microsoft/TypeScript/issues/19458 for verification steps. Basically, in a file that requires `sass.js`, confirm that you can get completions in a timely manner. Previously completions blocked until `geterr` returned, which could take several seconds

**Comment by isidorn** [comment]
I actually still get this with the latest insiders
I am using vscode ts version 2.9.1
![slow](https://user-images.githubusercontent.com/1926584/40660565-a682bf82-6351-11e8-80f0-bcbdbe5978af.gif)

**Comment by mjbvz** [comment]
Yeah, the full fix for this looks like it is move to TS 3.0

To verify the specific fix on our side: 

- Set 	`"typescript.tsserver.trace": "verbose"`
- Reload vs code
- Open the file with the sass.js require and quickly type something to trigger completions
- Open the output panel and go to typescript section
- Look for the `getErr` request number:

```
[Trace  - 11:14:40 PM] Sending request: geterr (55). Response expected: yes. Current queue length: 0
Arguments: {
    "delay": 0,
    "files": [
        "/Users/matb/projects/san/sub/foo.js"
    ]
}
```

- Look for the `geterr` request being cancelled when a `change` event happens:

```
[Trace  - 11:14:40 PM] Sending request: change (56). Response expected: no. Current queue length: 0
Arguments: {
    "file": "/Users/matb/projects/san/sub/foo.js",
    "line": 3,
    "offset": 4,
    "endLine": 3,
    "endOffset": 4,
    "insertString": "a"
}
[Trace  - 11:14:40 PM] TypeScript Service: trying to cancel ongoing request with sequence number 55
```
